lib/msal-common/src/telemetry/TelemetryManager.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { SERVER_TELEM_CONSTANTS, HeaderNames, CacheSchemaType } from "../utils/Constants";
+import { CacheManager } from "../cache/CacheManager";
+import { RequestFailures } from "./RequestFailures";
+import { AuthError } from "../error/AuthError";
+
+export class TelemetryManager {
+    private cacheManager: CacheManager;
+    private apiId: number;
+    private correlationId: string;
+    private forceRefresh: boolean;
+
+    constructor(cacheManager: CacheManager, apiId: number, correlationId: string, forceRefresh?: boolean) {
+        this.cacheManager = cacheManager;
+        this.apiId = apiId;
+        this.correlationId = correlationId;
+        this.forceRefresh = forceRefresh || false;
+    }
+
+    // API to add MSER Telemetry to request
+    protected currentRequest(): string {
+        const forceRefreshInt = this.forceRefresh ? 1 : 0;
+        const request = `${this.apiId}${SERVER_TELEM_CONSTANTS.VALUE_SEPARATOR}${forceRefreshInt}`;
+        const platformFields = ""; // TODO: Determine what we want to include
+
+        return [SERVER_TELEM_CONSTANTS.SCHEMA_VERSION, request, platformFields].join(SERVER_TELEM_CONSTANTS.CATEGORY_SEPARATOR);
```

Comment:
Should you be checking that the headers are not too large? I think the maximum size is recommended in the spec.

Reply:
Instead of checking the header size we're limiting ourselves to the last 3 errors which we're reasonably certain will never cause us to reach the limit

Reply:
Ok that should also work.